### PR TITLE
DSA : Recurrance Relation under issue #483 

### LINKED ISSUE #483: DSA 1.2.2.2 Algorithms: Recurrence Relations
## Description
Discuss Asymptotic Notations. What are Lower and Upper bounds? Describe their usages.

## Task
- [ ] Documentation
- [ ] Audio
- [ ] Video

## Note
- While requesting the issue to be assigned, please mention your **NAME & BATCH NUMBER**.
- Kindly refer to the [Issue Guidelines](https://github.com/girlscript/winter-of-contributing/blob/DSA/DSA/CONTRIBUTING.md#issues).
- In case you see that the issue has already been assigned, try to request for a different task within the same issue.
- Changes should be made inside the ````DSA/directory```` & ````DSA```` branch.
- Name of the file should be same as the issue.
- This issue is only for 'GWOC' contributors of 'DSA' Domain.

### PR DESCRIPTION
**Pull request by geeky01adarsh**
<hr>

## Description 📜

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. 
Fixes #483 
Added a md file in which I have described Asymptotic Notation and lower and upper bound. Please have a look @nimisha-yadav 
<hr>

## Type of change 📝

- [ ] Audio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Video (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [x] Documentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [ ] Android Dev (Flutter, Java, Kotlin)
- [ ] Backend Dev (Java, .NET, Python)
- [ ] C/CPP
- [ ] Competitive Programming
- [ ] Cyber Security
- [x] DSA
- [ ] Database
- [ ] Datascience with Python
- [ ] Datascience with R
- [ ] Frontend Dev (HTML, CSS, JS)
- [ ] Frontend Dev (React, Angular, Vue)
- [ ] Golang
- [ ] Interview Prep
- [ ] Java
- [ ] JavaScript
- [ ] MERN
- [ ] Machine Learning
- [ ] Open Source
- [ ] Python
- [ ] Rust
- [ ] Statistics
- [ ] UI/UX

<hr>
 
## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I'm GWOC'21 contributor

<hr>

### CODE REVIEW

**Review by geeky01adarsh — COMMENTED** [review]
I have self-reviewed the code and rechecked all spelling and grammatical errors. Please check it and let me know if any changes are required.

**Review by nimisha-yadav — COMMENTED** [review]
Please review @harshsri28

File: DSA/1.2 Analysis of Algorithms/Recurrence_Relations.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+
+### Usage of upper and lower bound
+> Let us understand this by an example. The very simple example of it could be bubble sort, the easiest sorting algo. If you don't know what it is, then expand the below region...
+<details>
+  <summary>What is Bubble Sort?</summary>
+  
+  ### Bubble Sort
+  > For every element of arr[], we check if arr[i]>arr[i+1], then we swap them. Bt this method, we are putting the largest element at last. We can improve the best case by maintaining a flag that will return 0, if arr[] is already sorted. Pseudo code for it would be.
+  ```
```
Comment (harshsri28) [inline DSA/1.2 Analysis of Algorithms/Recurrence_Relations.md:57]: Add funtion name 
eg ->
 ```
void bubbleSort( ){

}
```
Rest is looking good

Comment (geeky01adarsh) [reply DSA/1.2 Analysis of Algorithms/Recurrence_Relations.md:57]: > Add function name
> eg ->
> 
> ```
> void bubbleSort( ){
> 
> }
> ```
> 
> Rest is looking good

Resolved sir, please check it now.

**Review by harshsri28 — APPROVED** [review]
Looking good to me.

**Review by nimisha-yadav — CHANGES_REQUESTED** [review]
Looks good. @geeky01adarsh just make the following changes:
Update fork
Put your file under Recurrence Relations
Rename your file 1.2.2.2_Asymptotic_Notation_Upper_Lower_Bounds.md

**Comment by geeky01adarsh** [comment]
@nimisha-yadav mam done these tasks. Thanks!
- [x] Update fork
- [x] Put your file under Recurrence Relations
- [x] Rename your file 1.2.2.2_Asymptotic_Notation_Upper_Lower_Bounds.md

**Review by nimisha-yadav — APPROVED** [review]
Congrats on getting your PR merged!

**Comment by nimisha-yadav** [comment]
@geeky01adarsh, please send this PR again. I did not see your name as a contributor to this file so I'm gonna merge again.

This time, strictly follow the name guideline 1.2.2.2_Asymptotic_Notation_Upper_Lower_Bounds.md
**Do not add extra md.**

**Comment by geeky01adarsh** [comment]
> @geeky01adarsh, please send this PR again. I did not see your name as a contributor to this file so I'm gonna merge again.
> 
> This time, strictly follow the name guideline 1.2.2.2_Asymptotic_Notation_Upper_Lower_Bounds.md
> **Do not add extra md.**

I had done the same and my name was there from the first list. I am sorry if I had made any mistake. Also, I had renamed the file by literally copy-pasting it. As asked I am open this PR again. Thank You!

**Pull request merged**